### PR TITLE
Implement support for building GCP images

### PR DESCRIPTION
**Pull request by djmitche**
My changes on top of @walac's implementation -- a followon from #472.

[Bug 1567166](https://bugzilla.mozilla.org/show_bug.cgi?id=1567166)

### CODE REVIEW

**Comment by walac** [comment]
Does this require #465 merged?

File: deploy.sh [hunk 1]
```diff
@@ -2,14 +2,19 @@
 
 export BUILD_TARGET=$1
 if [ -z "$BUILD_TARGET" ]; then
-    echo "USAGE: ./deploy.sh <target>   (base, app, etc.)" >&2
+    echo "USAGE: ./deploy.sh <target> [options]  (base, app, etc.)" >&2
+    echo " --test -- (for legacy deployment only) only deploy to ami-test, not all workers"
+    ehco " --only='..' -- only build these packer builders"
```
Comment (walac) [inline deploy.sh]: +1

Comment (walac) [inline deploy.sh]: s/`ehco`/`echo`
Also would be nice if the help message provided the options, and how should we pass them (space or comma separated?)

**Comment by djmitche** [comment]
> Does this require #465 merged?

Nope.  *I* would like that merged, but the two PRs are independent :)

File: deploy.sh [hunk 2]
```diff
@@ -51,6 +68,23 @@ if ! $found; then
 fi
 echo "Using AWS account $AWS_ACCOUNT" >&2
 
+export GCP_PROJECT_ID
+case "$DEPLOYMENT" in
+    #taskcluster-net) GCP_PROJECT_ID=linux64-builds;;
```
Comment (walac) [inline deploy.sh]: I assume this is commented intentionally.

File: deploy.sh [hunk 3]
```diff
@@ -51,6 +68,23 @@ if ! $found; then
...
+    dustin-dev) GCP_PROJECT_ID=dustin-dev-3;;
+    *)
+        echo "No GCP project defined for this environment" >&2
+        exit 1;;
+esac
+
+gcloud projects describe $GCP_PROJECT_ID >/dev/null
+if [ $? != 0 ]; then
```
Comment (walac) [inline deploy.sh]: nit: we can shorten this with:
```sh
if !gcloud projects describe $GCP_PROJECT_ID >/dev/null; then
```

File: deploy/bin/build [hunk 4]
```diff
@@ -123,6 +138,7 @@ esac
 if [ -f packer-artifacts.json ]; then
   jq '
     .builds[]
+    | select(.builder_type == "amazon-ebs")
```
Comment (walac) [inline deploy/bin/build]: I don't know what I had in mind what I thought using `jq` would be a good idea.

File: deploy.sh (hunk 1 above)
Comment (djmitche) [reply deploy.sh]: Good point.  I was thinking of actually switching this to something like `--cloud=aws` to only build specific clouds, and then translating that to packer builders later.

File: deploy.sh (hunk 2 above)
Comment (djmitche) [reply deploy.sh]: It was, but probably unnecessarily.

File: deploy/bin/build (hunk 4 above)
Comment (djmitche) [reply deploy/bin/build]: It works fine, it just doesn't fit in my brain :)

**Comment by djmitche** [comment]
Hah, now I can't push to your branch..

**Pull request closed without merging**